plugins/workspace-minimap/test/minimap_tests.mocha.js

```diff
@@ -0,0 +1,377 @@
+      RTL: true,
+      horizontalLayout: true,
+      toolboxPosition: 'start',
+      move: {scrollbars: true},
+      scrollbar: {
+        isVisible() {
+          return true;
+        },
+        canScrollVertically() {
+          return true;
+        },
+        canScrollHorizontally() {
+          return true;
+        },
+      },
+      getMetrics() {
+        return {viewWidth: 100};
+      },
+    };
+  });
+
+  test('LTR Vertical Start', function() {
+    Object.assign(this.options, {
+      RTL: false,
+      horizontalLayout: false,
+      toolboxPosition: 'start',
+    });
+    Object.assign(this.mockMetrics, {
+      absoluteMetrics: {top: 0, left: 107},
+      toolboxMetrics: {position: Blockly.TOOLBOX_AT_LEFT},
+    });
+
+    const minimap = new PositionedMinimap(this.options);
+    minimap.setSize();
+    minimap.setPosition(this.mockMetrics, []);
+    const position = minimap.getBoundingRectangle();
+
+    assert.equal(position.top, 20, 'LTR Vertical Start: Incorrect top');
+    assert.equal(position.left, 872, 'LTR Vertical Start: Incorrect left');
+  });
+
+  test('LTR Vertical End', function() {
+    Object.assign(this.options, {
+      RTL: false,
+      horizontalLayout: false,
+      toolboxPosition: 'end',
+    });
+    Object.assign(this.mockMetrics, {
+      absoluteMetrics: {top: 0, left: 0},
+      toolboxMetrics: {position: Blockly.TOOLBOX_AT_RIGHT},
+    });
+
+    const minimap = new PositionedMinimap(this.options);
+    minimap.setSize();
+    minimap.setPosition(this.mockMetrics, []);
+    const position = minimap.getBoundingRectangle();
+
+    assert.equal(position.top, 20, 'LTR Vertical End: Incorrect top');
+    assert.equal(position.left, 20, 'LTR Vertical End: Incorrect left');
+  });
+
+  test('LTR Horizontal Start', function() {
+    Object.assign(this.options, {
+      RTL: false,
+      horizontalLayout: true,
+      toolboxPosition: 'start',
+    });
+    Object.assign(this.mockMetrics, {
+      absoluteMetrics: {top: 35, left: 0},
+      toolboxMetrics: {position: Blockly.TOOLBOX_AT_TOP},
+    });
+
+    const minimap = new PositionedMinimap(this.options);
+    minimap.setSize();
+    minimap.setPosition(this.mockMetrics, []);
+    const position = minimap.getBoundingRectangle();
+
+    assert.equal(position.top, 55, 'LTR Horizontal Start: Incorrect top');
+    assert.equal(position.left, 765, 'LTR Horizontal Start: Incorrect left');
+  });
+
+  test('LTR Horizontal End', function() {
+    Object.assign(this.options, {
+      RTL: false,
+      horizontalLayout: true,
+      toolboxPosition: 'end',
+    });
+    Object.assign(this.mockMetrics, {
+      absoluteMetrics: {top: 0, left: 0},
+      toolboxMetrics: {position: Blockly.TOOLBOX_AT_BOTTOM},
+    });
+
+    const minimap = new PositionedMinimap(this.options);
+    minimap.setSize();
+    minimap.setPosition(this.mockMetrics, []);
+    const position = minimap.getBoundingRectangle();
+
+    assert.equal(parseInt(position.top), 831,
```

Comment:
Why `parseInt`?

Reply:
Because the result is actually 831.66666666666

Reply:
Math.round() might make the intent a bit clearer.